dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/URLConnectionHTTPConduit.java

```diff
@@ -472,7 +475,17 @@ protected boolean usingProxy() {
         protected void setFixedLengthStreamingMode(int i) {
             // [CXF-6227] do not call connection.setFixedLengthStreamingMode(i)
             // to prevent https://bugs.openjdk.java.net/browse/JDK-8044726
+        
+            // Liberty Change for issue #25866-unexpected_EOF_from_server
+            // The bug [CXF-6227] is a performance bug observed for Java 7
+            // In case of a performance drop is observed need to create a walk around
+            boolean isRestMessage = PropertyUtils.isTrue(outMessage.getExchange().get(org.apache.cxf.message.Message.REST_MESSAGE));
```

Comment:
Is it possible for getExchange() to ever return null?   If so please check to avoid NPE.